File: wfe-restapi/src/main/java/ru/runa/wfe/rest/dto/WfeReport.java
package ru.runa.wfe.rest.dto;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class WfeReport {
    private Long id;
    private Long version;
    private String name;
    private String description;
    private String category;
    private List<WfeReportParameter> parameters = new ArrayList<>();
}

File: wfe-spa/src/ts/WfeReport.ts
import { WfeReportParameter } from './WfeReportParameter';

export class WfeReport {
    id: number|null = null;
    name: string = '';
    description: string = '';
    category: string = '';
    parameters: Array<WfeReportParameter>|null = null;

    getInfo(): object {
        return {
            'Название': this.name,
            'Описание': this.description
        };
    }
}